### PR TITLE
Replaced mapbox to osmdroid (Upload Activity)

### LINKED ISSUE #5414: More places where we need to move away from Mapbox maps?
I happened to use the app today with the changes for #5372. The Nearby screen appears not be using Mapbox tiles anymore. The other maps that are present in the app still seem to Mapbox tiles. This includes the following screen (as far as I've checked):

1. Explore -> Map
2. The edit location screen that appears when tapping on the map icon in first page of the upload flow.
3. Anything else?

Though they would have relatively fewer requests compared to the Nearby screen, they don't show appropriate Mapbox attribution now. So, we should either switch to OSM tiles in those screens or give appropriate Mapbox attribution.

PS: We ultimately need to start using Wikimedia maps tiles. That's the topic for #1002, though.

### PR DESCRIPTION
**Pull request by kanahia1**
**Description (required)**

Fixes #5414 

**Tests performed (required)**

Tested betaDebug on Samsung S21 FE with API level 33.

**Screenshots (for UI changes only)**
https://github.com/commons-app/apps-android-commons/assets/114223204/69139492-02bf-4f35-8ff1-c0ec46871c85

### CODE REVIEW

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It works great, except that in the Image Location activity (the activity that appears when you tap the Media Details' map icon) the pin should be fixed to the map. The pin should be movable to a different latitude/longitude only in the `Edit Location` activity.
I added a few other minor observations. :-)
Thanks!

File: app/src/main/res/drawable/ic_location_marker.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (kanahia1) [reply app/src/main/res/drawable/ic_location_marker.xml:1]: Sure I will reuse the same, Actually I was trying to recreate similar design present earlier in mapbox.

**Comment by kanahia1** [comment]
Hi @nicolas-raoul , I have made all the suggested changes ;-).  Should we change the color of marker in location picker similar to https://github.com/commons-app/apps-android-commons/pull/5440 ?

https://github.com/commons-app/apps-android-commons/assets/114223204/6a6449b7-12ce-494d-a459-a96fe817839f

**Comment by nicolas-raoul** [comment]
Your screencast looks great!
It is true that the position marker is a bit hard to see in dark mode, would you mind changing it? Thanks a lot!

**Review by nicolas-raoul — APPROVED** [review]

**Comment by kanahia1** [comment]
Thank you ;) for feedback @nicolas-raoul , I have changed the color of selected location marker https://github.com/commons-app/apps-android-commons/pull/5443/commits/9814bfe51caf5f2ca1d392b2a2a39671c5c0dc29

WhatsApp Image 2024-01-17 at 15 35 07_4b6e5e07

**Pull request merged**